### PR TITLE
Simple animals can attack/break cameras

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
rscadd: Simple animals may now attack and destroy cameras.
/:cl:

### CODE REVIEW

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -129,6 +129,18 @@
...
+	if(!istype(user))
+		return
+	if(user.melee_damage_upper >= 10)
+		if(user.attack_sound)
+			playsound(src.loc, user.attack_sound, 100, 1)
+		user.do_attack_animation(src)
+		user.changeNext_move(CLICK_CD_MELEE)
+		health = max(0, health - user.melee_damage_upper)
+		if(!health && status)
+			deactivate(user, 0)
```
Comment (GunHog) [inline code/game/machinery/camera/camera.dm]: `triggerCameraAlarm()` here, just like it is for when humans bash cameras with high force objects.

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -129,6 +129,18 @@
 	if(!health && status)
 		deactivate(user, 0)
 
+/obj/machinery/camera/attack_animal(mob/living/simple_animal/user)
+	if(!istype(user))
+		return
+	if(user.melee_damage_upper >= 10)
```
Comment (ChangelingRain) [inline code/game/machinery/camera/camera.dm]: kor, why. generate the damage from upper and lower, then check if it's above 10 to see if you damage the camera(and insert appropriate feedback if it fails to do enough damage)

Comment (KorPhaeron) [inline code/game/machinery/camera/camera.dm]: RNG damage was dumb anyway I regret doing the upper/lower thing

Comment (KazeEspada) [inline code/game/machinery/camera/camera.dm]: Why 10? I mean that's a lot as far as animals go. None of the pets have above 9. Also, you saying poly can't cut a camera?

Comment (KorPhaeron) [inline code/game/machinery/camera/camera.dm]: That is the force threshold for attacking it with objects

**Comment by WJohn** [comment]
I hope simple animals without players won't be able to passively whack at them, because this'd be a gigantic buff to the staff of animation.

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -129,6 +129,19 @@
...
+		if(user.attack_sound)
+			playsound(src.loc, user.attack_sound, 100, 1)
+		user.do_attack_animation(src)
+		user.changeNext_move(CLICK_CD_MELEE)
+		health = max(0, health - user.melee_damage_upper)
+		if(!health && status)
+			triggerCameraAlarm()
+			deactivate(user, 1)
```
Comment (tkdrg) [inline code/game/machinery/camera/camera.dm]: Most of the latter half of this proc is copy-pasted from attackby. Please offload it to its own proc (something like `loseHealth(mob/living/user, damage)` would do) and call it on both.

Comment (tkdrg) [inline code/game/machinery/camera/camera.dm]: @KorPhaeron

**Comment by KorPhaeron** [comment]
Gonna be easier to rewrite one paragraph than fix a conflict on this since proc names changed and such

**Pull request closed without merging**